pyaedt/application/Design.py

```diff
@@ -3921,3 +3921,34 @@ def set_temporary_directory(self, temp_dir_path):
         """
         self.odesktop.SetTempDirectory(temp_dir_path)
         return True
+
+    @pyaedt_function_handler()
+    def design_settings(self):
+        """Get design settings for the current AEDT app.
+
+        Returns
+        -------
+        dict
+            Dictionary of valid design settings.
+
+        References
+        ----------
+
+        >>> oDesign.GetChildObject("Design Settings")
+        """
+        try:
+            design_settings = self._odesign.GetChildObject("Design Settings")
+        except Exception:  # pragma: no cover
+            self.logger.error("Could not execute GetChildObject('Design Settings').")
+            return False
+
+        prop_name_list = design_settings.GetPropNames()
+        design_settings_dict = {}
+        for prop in prop_name_list:
+            try:
+                design_settings_dict[prop] = design_settings.GetPropValue(prop)
+            except Exception:  # pragma: no cover
+                self.logger.warning("Could not retrieve property value in design settings")
+                design_settings_dict[prop] = "Could not retrieve the value."
```

Comment:
`None` might be better than a string when the property is not retrieved. In this way one can directly evaluate the dictionary value with `if design_settings_dict[prop]` to check whether the setting was correctly loaded.